src/endpoints/games.js

```diff
@@ -98,27 +98,81 @@ function _getPGN(pgnHeaders, moveList) {
     chess.header(key, value);
   }
   for (let i = 0; i < moveListLength; i += 2) {
-    let encFrom = moveList[i];
-    let encTo = moveList[i + 1];
-    chess.move({
-      'from': _decodeMove(encFrom),
-      'to': _decodeMove(encTo)
-    });
+    let move = {
+      'from': _decodeMove(moveList[i], false),
+      'to': _decodeMove(moveList[i + 1], true)
+    }
+    if (move.to instanceof _PawnPromotion) {
+      move.promotion = move.to.piece;
+      move.to = move.to.getTo(move.from);
+    }
+    chess.move(move);
   }
   return chess.pgn();
 }
 
-/* Decode a move-character into algebraic notation */
-function _decodeMove(encMove) {
+/* Decode a move-character into algebraic notation or pawn promotion */
+function _decodeMove(encMove, isTo) {
   const index = BOARD_POSITIONS.indexOf(encMove);
   if (index === -1) {
+    // if this is the "to" field, check for pawn promotion
+    if (isTo) {
+      const promotionIndex = PROMOTION_TABLE.indexOf(encMove);
+      if (promotionIndex !== -1) {
+        return new _PawnPromotion(promotionIndex);
+      }
+    }
     throw new Error(`Unrecognized move-character: ${encMove}`);
   }
   const file = BOARD_FILES[index % BOARD_LENGTH];
   const rank = Math.floor(index / BOARD_LENGTH) + 1;
   return `${file}${rank}`;
 }
 
+/* _PawnPromotion constructor */
+function _PawnPromotion(index) {
+  const pieceIndex = Math.floor(index / PROMOTION_TABLE_COLUMNS_LENGTH);
+  if (pieceIndex > PROMOTION_TABLE_ROWS_LENGTH - 1) {
+    // this can only happen if the const table values are wrong
+    throw new Error(`Pawn promotion row index out of bounds: ${pieceIndex}`);
+  }
+  this.piece = PROMOTION_TABLE_ROWS[pieceIndex];
+  this.pieceIndex = index;
+  this.isCaptureLeft = index % PROMOTION_TABLE_COLUMNS_LENGTH 
+      === PROMOTION_CAPTURE_LEFT;
+  this.isCaptureRight = index % PROMOTION_TABLE_COLUMNS_LENGTH
+      === PROMOTION_CAPTURE_RIGHT;
+}
+
+/* Calculate the destination square of a pawn promotion */
+_PawnPromotion.prototype.getTo = function(from) {
+  const fromFile = from[0];
+  const fromRank = from[1];
+  const fromFileIndex = BOARD_FILES.indexOf(fromFile);
+  const toFileIndex = this.isCaptureLeft ?
+      fromFileIndex - 1 :
+      this.isCaptureRight ? 
+          fromFileIndex + 1 :
+          fromFileIndex;
+
+  // sanity check: ensure pawn is still on the board after promotion
+  if (toFileIndex < 0 || toFileIndex > BOARD_LENGTH - 1) {
+    throw new Error(`Invalid pawn promotion; file index out of bounds: ${toFileIndex}`);
+  }
+  const toFile = BOARD_FILES[toFileIndex];
+
+  // sanity check: ensure pawn rank is 2 or 7 prior to promotion
```

Comment:
Good sanity check here. Possible also future on that makes sure new piece is on rank 1 or 8